src/vs/workbench/services/configurationResolver/browser/configurationResolverService.ts

```diff
@@ -197,46 +195,69 @@ export abstract class BaseConfigurationResolverService extends AbstractVariableR
 			}
 
 			if (typeof result === 'string') {
-				variableValues[variable] = result;
+				return result;
 			} else {
 				return undefined;
 			}
-		}
+		});
 
 		return variableValues;
 	}
 
 	/**
 	 * Recursively finds all command or input variables in object and pushes them into variables.
 	 * @param object object is searched for variables.
-	 * @param variables All found variables are returned in variables.
+	 * @param variables All found variables and resolved values are returned in variables.
+	 * @param resolveFn callback method for resolving the variable value.
+	 * @param source the root level object for passthrough to the resolveFn.
 	 */
-	private findVariables(object: any, variables: string[]) {
+	private async findVariables(object: any, variables: IStringDictionary<string>, resolveFn: (variable: string, root: any) => Promise<string | undefined>, source?: any): Promise<any> {
+
+		const root: any = source || object;
 		if (typeof object === 'string') {
 			let matches;
-			while ((matches = BaseConfigurationResolverService.INPUT_OR_COMMAND_VARIABLES_PATTERN.exec(object)) !== null) {
+			const input = object;
+			while ((matches = BaseConfigurationResolverService.INPUT_OR_COMMAND_VARIABLES_PATTERN.exec(input)) !== null) {
 				if (matches.length === 4) {
+					const entry = matches[0];
 					const command = matches[1];
-					if (variables.indexOf(command) < 0) {
-						variables.push(command);
+					const found = command in variables;
+					if (!found) {
+						// NOTE this if assumes that the user will only ever want one computed value. (e.g. have two of the same command but only get the first output for both)
```

Comment:
This assumption was from the existing code.  Just added a comment to make it clear while I was making changes.